src/backend/distributed/citus--8.3-1--8.4-1.sql

```diff
@@ -0,0 +1,95 @@
+/* citus--8.3-1--8.4-1 */
+
+/* bump version to 8.4-1 */
+CREATE OR REPLACE FUNCTION pg_catalog.citus_prepare_pg_upgrade()
+    RETURNS void
+    LANGUAGE plpgsql
+    SET search_path = pg_catalog
+    AS $cppu$
+BEGIN
+    -- backup citus catalog tables
+    CREATE TABLE public.pg_dist_authinfo AS SELECT * FROM pg_catalog.pg_dist_authinfo;
+    CREATE TABLE public.pg_dist_colocation AS SELECT * FROM pg_catalog.pg_dist_colocation;
+    CREATE TABLE public.pg_dist_local_group AS SELECT * FROM pg_catalog.pg_dist_local_group;
+    CREATE TABLE public.pg_dist_node AS SELECT * FROM pg_catalog.pg_dist_node;
+    CREATE TABLE public.pg_dist_node_metadata AS SELECT * FROM pg_catalog.pg_dist_node_metadata;
+    CREATE TABLE public.pg_dist_partition AS SELECT * FROM pg_catalog.pg_dist_partition;
+    CREATE TABLE public.pg_dist_placement AS SELECT * FROM pg_catalog.pg_dist_placement;
+    CREATE TABLE public.pg_dist_poolinfo AS SELECT * FROM pg_catalog.pg_dist_poolinfo;
+    CREATE TABLE public.pg_dist_shard AS SELECT * FROM pg_catalog.pg_dist_shard;
+    CREATE TABLE public.pg_dist_transaction AS SELECT * FROM pg_catalog.pg_dist_transaction;
+END;
+$cppu$;
+
+COMMENT ON FUNCTION pg_catalog.citus_prepare_pg_upgrade()
+    IS 'perform tasks to copy citus settings to a location that could later be restored after pg_upgrade is done';
+
+
+CREATE OR REPLACE FUNCTION pg_catalog.citus_finish_pg_upgrade()
+    RETURNS void
+    LANGUAGE plpgsql
+    SET search_path = pg_catalog
+    AS $cppu$
+DECLARE
+    table_name regclass;
+    command text;
+    trigger_name text;
+BEGIN
+    -- restore citus catalog tables
+    INSERT INTO pg_catalog.pg_dist_authinfo SELECT * FROM public.pg_dist_authinfo;
```

Comment:
Minor: I think the order or these statements doesn't matter at the moment. Still, I like the order that is on the documentation for some related tables, because if we one day add foreign keys between metadata tables, that'd be robust. 

(Also, the flow of that feels/sounds more natural for `pg_dist_partition ` -> `pg_dist_shard ` -> `pg_dist_placement ` ) 

In any case, I think it'd be nice to bucket enterprise-only tables at the end would be easier to read/maintain.

Reply:
Your reasoning makes sense, didn't take FK's into account. For these blocks to be easy to maintain I always sort them alphabetically. Now with your reasoning I will change the order back to the order in the docs and separate the enterprise catalogs to the bottom.

Reply:
I think fixing the order makes sense. 

Small note on foreign keys: The one foreign key we added (pg_dist_poolinfo -> pg_dist_node) is causing some issues. pg_dist_poolinfo gets truncated during metadata syncing, but it's supposed to be managed by the control plane. We should probably remove this key.